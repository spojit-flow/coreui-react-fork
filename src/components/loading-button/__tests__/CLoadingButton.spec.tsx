import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CLoadingButton } from '../../../index'

test('loads and displays CLoadingButton component', async () => {
  const { container } = render(<CLoadingButton />)
  expect(container).toMatchSnapshot()
})

test('CLoadingButton customize', async () => {
  const { container } = render(
    <CLoadingButton
      className="bazinga"
      disabledOnLoading={true}
      loading={true}
      spinnerType="grow"
      timeout={1000}
    >
      Text
    </CLoadingButton>,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('btn-loading')
  expect(container.firstChild).toHaveClass('is-loading')
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveAttribute('disabled', '')
  expect(container.firstChild).toHaveTextContent('Text')
  if (container.firstChild === null) {
    expect(true).toBe(false)
  } else {
    expect(container.firstChild.firstChild).toHaveClass('btn-loading-spinner')
    expect(container.firstChild.firstChild).toHaveClass('spinner-grow')
    expect(container.firstChild.firstChild).toHaveClass('spinner-grow-sm')
  }
})

test('CLoadingButton test event on click', async () => {
  const onClick = jest.fn()
  render(<CLoadingButton onClick={onClick}>Text</CLoadingButton>)
  expect(onClick).toHaveBeenCalledTimes(0)
  const button = document.querySelector('.btn-loading')
  if (button !== null) {
    fireEvent.click(button)
  }
  expect(onClick).toHaveBeenCalledTimes(1)
})
