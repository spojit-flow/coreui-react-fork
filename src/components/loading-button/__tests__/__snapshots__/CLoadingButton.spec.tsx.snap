// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLoadingButton customize 1`] = `
<div>
  <button
    class="btn btn-primary btn-loading is-loading bazinga disabled"
    disabled=""
    type="button"
  >
    <div
      class="spinner-grow text-undefined spinner-grow-sm btn-loading-spinner"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
    Text
  </button>
</div>
`;

exports[`loads and displays CLoadingButton component 1`] = `
<div>
  <button
    class="btn btn-primary btn-loading"
    type="button"
  >
    <div
      class="spinner-border text-undefined spinner-border-sm btn-loading-spinner"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </button>
</div>
`;
